src/components/button/button.fluent.scss

```diff
@@ -0,0 +1,143 @@
+@use '../../styles/utilities' as utils;
+
+@mixin theme() {
+    igc-button[variant='raised']::part(base),
+    igc-link-button[variant='raised']::part(base),
+    igc-button[variant='fab']::part(base),
+    igc-link-button[variant='fab']::part(base) {
+        box-shadow: var(--igc-elevation-0);
+    }
+
+    igc-button[size='small']::part(base),
+    igc-link-button[size='small']::part(base) {
+        border-radius: calc(var(--igc-border-radius) * 1);
+        --size: #{utils.rem(24px)};
+    }
+
+    igc-button[size='medium']::part(base),
+    igc-link-button[size='medium']::part(base) {
+        border-radius: calc(var(--igc-border-radius) * 1);
+        --size: #{utils.rem(28px)};
+    }
+
+    igc-button[size='large']::part(base),
+    igc-link-button[size='large']::part(base) {
+        border-radius: calc(var(--igc-border-radius) * 1);
+        --size: #{utils.rem(32px)};
+    }
+
+    // FLAT
+    igc-button[variant='flat']:not([disabled])::part(base),
+    igc-link-button[variant='flat']:not([disabled])::part(base) {
+        color: #{utils.color(gray, 900)};
+
+        &:hover {
+            color: #{utils.color(primary, 500)};
+
+            &::before {
+                background: transparent;
+            }
+        }
+
+        &:focus,
+        &:active {
+            color: #{utils.color(primary, 500)};
+
+            &::before {
+                opacity: .8;
+                background: #{utils.color(gray, 100)};
```

Comment:
Is there any reason not to apply the opacity directly to the color as the third parameter?